compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/writer/BIRInstructionWriter.java

```diff
@@ -138,6 +143,9 @@ public void visit(BIRBasicBlock birBasicBlock) {
 
         // write pos and kind
         writePosition(terminator.pos);
+        if (terminator.kind != InstructionKind.RETURN) {
+            writeScopes(terminator);
```

Comment:
Can we move if condition inside writeScopes method?

Reply:
Addressed with https://github.com/ballerina-platform/ballerina-lang/pull/31829/commits/111094729f69d3296a3c694581dabfdebc283d93